Task4/controller/src/main/java/com/epam/esm/security/JwtUserDetailsService.java

```diff
@@ -0,0 +1,28 @@
+package com.epam.esm.security;
+
+import com.epam.esm.entity.User;
+import com.epam.esm.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Component;
+
+@Component
+public class JwtUserDetailsService implements UserDetailsService {
+    private UserService userService;
+
+    @Autowired
+    public void setUserService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String s) {
+        User user = userService.loadUserByUsername(s);
+        if (user == null) {
```

Comment:
Try to use Objects.isNull(user) and follow this rule along the code.
Everywhere, where you can use internal functions - try to use it - best practice is to write code with the least amount of operands.